### PR TITLE
feat(amazon q): adding support to accept java 17 projects

### PR DESCRIPTION
**Pull request by ntarakad-aws**
## Types of changes

New Feature

## Description

Enabling support for java 17 projects
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rli — APPROVED** [review]
is the idea supporting rewrites into modern syntax?

**Comment by dhasani23** [comment]
Waiting to approve once we complete the 1) string updates and 2) E2E testing

File: plugins/core/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties [hunk 1]
```diff
@@ -707,11 +707,11 @@ codemodernizer.notification.warn.download_failed_other.content=Amazon Q ran into
...
+codemodernizer.notification.warn.invalid_project.description.reason.no_valid_files=None of your open modules are supported for code transformation with Amazon Q. A pom.xml is required for transformation. Amazon Q can upgrade Java 8, Java 11, and Java 17 projects built on Maven.
 codemodernizer.notification.warn.invalid_project.description.reason.not_logged_in=Amazon Q cannot start the transformation as you are not logged in with Identity Center or Builder ID. Also ensure that you are not using IntelliJ version 232.8660.185 and that you are not developing on a remote host (uncommon).
-codemodernizer.notification.warn.invalid_project.description.reason.remote_backend=None of your open modules are supported by Amazon Q Code Transformation. Currently, Amazon Q can only upgrade Java 8 or Java 11 projects built on Maven, and cannot upgrade projects running on a remote host.
+codemodernizer.notification.warn.invalid_project.description.reason.remote_backend=None of your open modules are supported for code transformation with Amazon Q. Amazon Q can't transform projects running on a remote host.
```
[the remark this answers is not in the document]
Comment (ntarakad-aws) [reply plugins/core/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:714]: Looks like we use "cannot" so I can change to this

File: plugins/core/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties [hunk 2]
```diff
@@ -632,9 +632,9 @@ codemodernizer.chat.message.upload_failed_url_expired=Sorry, I couldn't upload y
...
-codemodernizer.chat.message.validation.error.unsupported_java_version=Sorry, I couldn't find a module that I can upgrade. Currently, I can only upgrade Java 8 or Java 11 modules built on Maven. If you have a Java 8 or Java 11 module in your workspace, you might need to configure your project so that I can find it. Go to File and choose Project Structure. In the Projects tab, set the correct project JDK and the correct language level. In the Modules tab, set the correct module JDK and language level.
+codemodernizer.chat.message.validation.error.no_pom=I couldn't find a module that I can upgrade. Your Java project must be built on Maven and contain a pom.xml file. For more information, see the Amazon Q documentation.
```
Comment (dhasani23) [inline plugins/core/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:635]: Typically when we say "Amazon Q documentation" we hyperlink to it, see line 634 above for example

File: plugins/amazonq/mynah-ui/src/mynah-ui/ui/quickActions/generator.ts [hunk 3]
```diff
@@ -37,7 +37,7 @@ export class QuickActionGenerator {
                         ? [
                               {
                                   command: '/transform',
-                                  description: 'Transform your Java 8 or 11 Maven project to Java 17',
+                                  description: 'Transform your Java Project',
```
Comment (dhasani23) [inline plugins/amazonq/mynah-ui/src/mynah-ui/ui/quickActions/generator.ts:40]: nit: lowercase project

**Review by dhasani23 — APPROVED** [review]

**Pull request merged**